packages/db/fauna/resources/Function/deleteUserUpload.js

```diff
@@ -35,20 +38,42 @@ const body = Query(
         Let(
           {
             content: Get(Var('contentMatch')),
-            uploadMatch: Match(
-              Index('upload_by_user_and_content'),
-              Var('userRef'),
-              Select('ref', Var('content')),
-              true
-            )
+            dagSize: Select(['data', 'dagSize'], Var('content'))
           },
           If(
-            IsNonEmpty(Var('uploadMatch')),
-            Update(
-              Select(['ref'], Get(Var('uploadMatch'))),
-              { data: { deleted: Now() } }
+            IsNumber(Var('dagSize')),
+            Let(
+              {
+                uploadMatch: Match(
+                  Index('upload_by_user_and_content'),
+                  Var('userRef'),
+                  Select('ref', Var('content')),
+                  true
+                ),
+                upload: Select(['ref'], Get(Var('uploadMatch')))
```

Comment:
You can't do this here because the match could be empty.

```suggestion
                )
```